pkg/providers/instance/instance.go

```diff
@@ -435,6 +462,14 @@ func (p *Provider) launchInstance(
 		return nil, nil, azErr
 	}
 
+	if p.provisionMode == options.ProvisionModeBootstrappingClient {
+		err = p.createCSExtension(ctx, resourceName, launchTemplate.AgentBakerCSE, launchTemplate.IsWindows)
```

Comment:
We have the aksNodeClass here we dont need launchTemplate.IsWindows we can just do something like 
```
		err = p.createCSExtension(ctx, resourceName, launchTemplate.AgentBakerCSE, lo.Ternary(nodeClass.Spec.ImageFamily == Windows), true, false)
```

we can add a helper too in the AKSNodeClass api that is called iswindows doing that check so we can reuse that everywhere we have access to the nodeclass